apps/web/pages/apps/onboarding/[[...step]].tsx

```diff
@@ -0,0 +1,502 @@
+  appNotExtendsEventType: "App does not extend EventTypes",
+  userNotInTeam: "User is not in provided team",
+} as const;
+
+const getUser = async (userId: number) => {
+  const user = await prisma.user.findUnique({
+    where: {
+      id: userId,
+    },
+    select: {
+      id: true,
+      avatar: true,
+      name: true,
+      username: true,
+      teams: {
+        select: {
+          accepted: true,
+          team: {
+            select: {
+              id: true,
+              name: true,
+              logo: true,
+            },
+          },
+        },
+      },
+    },
+  });
+
+  if (!user) {
+    throw new Error(ERROR_MESSAGES.userNotFound);
+  }
+  return user;
+};
+
+const getAppBySlug = async (appSlug: string) => {
+  const app = await prisma.app.findUnique({
+    where: { slug: appSlug, enabled: true },
+    select: { slug: true, keys: true, enabled: true, dirName: true },
+  });
+  if (!app) throw new Error(ERROR_MESSAGES.appNotFound);
+  return app;
+};
+
+const getEventTypes = async (userId: number, teamId?: number) => {
+  const eventTypes = (
+    await prisma.eventType.findMany({
+      select: {
+        id: true,
+        description: true,
+        durationLimits: true,
+        metadata: true,
+        length: true,
+        title: true,
+        position: true,
+        recurringEvent: true,
+        requiresConfirmation: true,
+        team: { select: { slug: true } },
+        slug: true,
+      },
+      where: teamId ? { teamId } : { userId },
+    })
+  ).sort((eventTypeA, eventTypeB) => {
+    return eventTypeB.position - eventTypeA.position;
+  });
+  if (eventTypes.length === 0) {
+    throw new Error(ERROR_MESSAGES.noEventTypesFound);
+  }
+  return eventTypes;
+};
+
+const getEventTypeById = async (eventTypeId: number) => {
+  const eventTypeDB = await prisma.eventType.findFirst({
+    select: {
+      id: true,
+      slug: true,
+      description: true,
+      users: { select: { username: true } },
+      length: true,
+      title: true,
+      teamId: true,
+      seatsPerTimeSlot: true,
+      recurringEvent: true,
+      team: { select: { slug: true } },
+      schedulingType: true,
+      metadata: true,
+    },
+    where: { id: eventTypeId },
+  });
+
+  if (!eventTypeDB) {
+    throw new Error(ERROR_MESSAGES.noEventTypesFound);
+  }
+  return {
+    ...eventTypeDB,
+    URL: `${CAL_URL}/${
+      eventTypeDB.team ? `team/${eventTypeDB.team.slug}` : eventTypeDB?.users?.[0]?.username
+    }/${eventTypeDB.slug}`,
+  } as ConfigureEventTypeProp;
+};
+
+const getAppInstallsBySlug = async (appSlug: string, userId: number, teamId?: number) => {
+  const appInstalls = await prisma.credential.findMany({
+    where: {
+      OR: [
+        {
+          appId: appSlug,
+          userId: userId,
+        },
+        teamId
+          ? {
+              appId: appSlug,
+              teamId: 5,
+            }
+          : {},
+      ],
+    },
+  });
+  return appInstalls;
+};
```

Comment:
all these functions should be in a Repositories on the api side, so that getServerSideProps doesn't depend on prisma directly